core/src/main/scala/org/apache/spark/partial/ApproximateActionListener.scala

```diff
@@ -34,11 +34,13 @@ private[spark] class ApproximateActionListener[T, U, R](
     rdd: RDD[T],
     func: (TaskContext, Iterator[T]) => U,
     evaluator: ApproximateEvaluator[U, R],
-    timeout: Long)
+    timeout: Long = 1000*60*60*24*30*12,
```

Comment:
@Ru-Xiang that's orthogonal to what @srowen was suggesting. The issue here is that the change is breaking binary compatibility for Spark applications.